app/models/manageiq/providers/google/resources.rb

```diff
@@ -0,0 +1,25 @@
+module ManageIQ::Providers::Google
+  class Resources
+    # Parses a Google resource url and returns the resource from the path.
+    # This method understands both complete urls (starting with
+    # https://www.googleapis.com) as well as partials.
+    def self.get_resource_type(uri)
+      # We can cheat here a bit - most the google APIs (at least in compute.v1)
+      # define resources directly under their type (i.e. snapshots/foo or
+      # images/bar). However, 'images' allow the specification of 'family' which
+      # is an alias to the latest image in a family, breaking the rule.
+      return nil if uri.nil? || uri == ''
+
+      parts = uri.split('/')
+      return nil if parts.length <= 1
+      return nil if parts[-2] == ''
```

Comment:
Same idea [as above](https://github.com/ManageIQ/manageiq/pull/7844/files#r59153685):
```ruby
return if parts.length <= 1
return if parts[-2] == ''
```

But, I believe the above two lines could just be:
```ruby
return if parts[-2].blank?
```

This handles the cases of:

* `uri == "single"`
* `uri == "/leadingslash"`

In the first case, `parts == ["single"]` and `parts[-2] == nil`, so `parts[-2].blank? == true`.

In the second case, `parts == ["", "leadingslash"]` and `parts[-2] == ""`, so `parts[-2].blank? == true`.

I might have a gap in there somewhere, but I think it's right.

@durandom, @agrare ... check me on that